src/class-wpml-beaver-builder-cleanup-hooks.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+class WPML_Beaver_Builder_Cleanup_Hooks implements IWPML_Action {
+
+	public function add_hooks() {
+		add_action( 'wpml_delete_unused_package_strings', array( $this, 'delete_block_layout_string' ) );
+	}
+
+	/**
+	 * @param array $package_data
+	 */
+	public function delete_block_layout_string( $package_data ) {
+		$packages = apply_filters( 'wpml_st_get_post_string_packages', array(), $package_data['post_id'] );
+
+		/** @var WPML_Package[] $packages */
+		foreach ( $packages as $package ) {
+			if ( WPML_Gutenberg_Integration::PACKAGE_ID === $package->kind ) {
+				$strings = $package->get_package_strings();
+
+				$is_unique_string = count( $strings ) < 2;
+
+				foreach ( $strings as $string ) {
+					if ( 'fl-builder/layout' === $string->title ) {
+
+						if ( $is_unique_string ) {
+							do_action( 'wpml_delete_package', $package->name, $package->kind );
+						} else {
+							do_action( 'wpml_st_delete_all_string_data', $string->id );
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
@andreasciamanna, here's an exemple of the content stored in the DB when I create a page with 1 heading module and 1 text module (with Beaver Builder front end's editor):

```
<!-- wp:fl-builder/layout --><h3>
        Good monrning2
    </h3>
    <p>Hello there</p>
<!-- /wp:fl-builder/layout -->
```

Beaver Builder wraps the content with the (gutenberg) block `fl-builder/layout`. The 2 modules are already parsed by our custom logic for Beaver Builder.

So we end up with 3 strings:
- 2 in a "Beaver Builder" package (heading module and text module).
- 1 in a "Gutenberg" package (the whole content).

As a consequence, we will have 3 translation fields instead of 2 in the translation job.

At the beginning, I also thought about deactivating our Gutenberg logic when Beaver Builder is enabled, but Beaver Builder still allows to use the (Gutenberg) block editor.

So the idea is to cleanup this layout string after the whole process.

Reply:
@strategio as I wrote, I have no issues understanding the problem you are fixing.

I have issues understanding your fix instead.
In particular, while I understand that you must clean up the string, I would expect that anything removed from that string will be restored at one point (e.g. when the translation is completed).

Reply:
We don't need to restore it @andreasciamanna. We just don't want the Gutenberg block `fl-builder/layout` to be handled by the WPML's Gutenberg parsing logic. The `fl-builder/layout` string (registered with WPML's Gutenberg parsing) is kind of duplicating the 2 other strings (registered with WPML's Beaver Builder parsing). Basically, this Gutenberg's block means that the wrapped content was built with Beaver Builder, and we already have all the logic to handle it.

We could also have just excluded this `fl-builder/layout` string at registering time (with a new filter), but we would not know if this is a unique package string or not. And this could result in empty string packages (ghost entries). That's why I prefer a final cleanup.